guardrails/run/async_stream_runner.py

```diff
@@ -0,0 +1,567 @@
+
+        Returns:
+            The Call log for this run.
+        """
+        if prompt_params is None:
+            prompt_params = {}
+
+        # check if validator requirements are fulfilled
+        missing_keys = verify_metadata_requirements(
+            self.metadata, self.output_schema.root_datatype
+        )
+        if missing_keys:
+            raise ValueError(
+                f"Missing required metadata keys: {', '.join(missing_keys)}"
+            )
+
+        (
+            instructions,
+            prompt,
+            msg_history,
+            prompt_schema,
+            instructions_schema,
+            msg_history_schema,
+            output_schema,
+        ) = (
+            self.instructions,
+            self.prompt,
+            self.msg_history,
+            self.prompt_schema,
+            self.instructions_schema,
+            self.msg_history_schema,
+            self.output_schema,
+        )
+
+        return self.async_step(
+            index=0,
+            api=self.api,
+            instructions=instructions,
+            prompt=prompt,
+            msg_history=msg_history,
+            prompt_params=prompt_params,
+            prompt_schema=prompt_schema,
+            instructions_schema=instructions_schema,
+            msg_history_schema=msg_history_schema,
+            output_schema=output_schema,
+            output=self.output,
+            call_log=call_log,
+        )
+
+    async def async_run(
+        self, call_log: Call, prompt_params: Optional[Dict] = None
+    ) -> Call:
+        if prompt_params is None:
+            prompt_params = {}
+
+        missing_keys = verify_metadata_requirements(
+            self.metadata, self.output_schema.root_datatype
+        )
+
+        if missing_keys:
+            raise ValueError(
+                f"Missing required metadata keys: {', '.join(missing_keys)}"
+            )
+
+        (
+            instructions,
+            prompt,
+            msg_history,
+            prompt_schema,
+            instructions_schema,
+            msg_history_schema,
+            output_schema,
+        ) = (
+            self.instructions,
+            self.prompt,
+            self.msg_history,
+            self.prompt_schema,
+            self.instructions_schema,
+            self.msg_history_schema,
+            self.output_schema,
+        )
+
+        result = self.async_step(
+            index=0,
+            api=self.api,
+            instructions=instructions,
+            prompt=prompt,
+            msg_history=msg_history,
+            prompt_params=prompt_params,
+            prompt_schema=prompt_schema,
+            instructions_schema=instructions_schema,
+            msg_history_schema=msg_history_schema,
+            output_schema=output_schema,
+            output=self.output,
+            call_log=call_log,
+        )
+        async for call in result:
+            # yield ValidationOutcome[OT].from_guard_history(call)
+            yield call
```

Comment:
These seem duplicative.

Reply:
This is essentially just an await from the async generator. It is still doing something, just pretty verbose. I will put a comment and clean it up unless we have a better place for doing this

Reply:
Sorry not this line specifically, I commented on a large block which Github doesn't show very well.  What I was saying was duplicative was `__call__` and `async_run`.  I don't believe we're using `__call__`.  I think long term `async_run` should become the override of `__call__` but for now I think we can just drop the `__call__` implementation like how `AsyncRunner` does.

Reply:
Yep, verified that it was unused and removed. Good catch